### PR TITLE
Cluster Map & Co.

### PR DESCRIPTION
**Pull request by ellsclytn**
Adds metrics to the side, and a cluster map!

### CODE REVIEW

File: source/components/ClusterMap/index.js [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    limit: 100,
+    page
+  }}).then(({ data }) => ({
+    pageCount: data.meta.total_pages,
+    pages: data.pages.map((page) => ({
+      id: page.id,
+      name: page.name,
+      coordinates: page.coordinate && [page.coordinate.lon, page.coordinate.lat],
+      amount: page.amount.cents
+    })).filter((page) => (
```
Comment (aabrook) [inline source/components/ClusterMap/index.js]: For an optimisation you could filter before mapping. Unnecessary of course as it's not a large dataset

File: source/components/ClusterMap/index.js [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+  componentDidMount () {
+    getPages().then(({ pageCount, pages }) => {
+      if (pageCount > 1) {
+        return Promise.all(Array(pageCount - 1).fill().map((_, i) => (
+          getPages(i + 2) // We need to skip page 1, it's already fetched
+        ))).then((pageResults) => (
+          pageResults.reduce((acc, { pages: pageSet }) => ([
+            ...acc,
+            ...pageSet
+          ]), [])
```
Comment (aabrook) [inline source/components/ClusterMap/index.js]: If you make the starting array pages you won't need the below then

File: source/components/ClusterMap/index.js [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+
+  render () {
+    if (typeof window === 'undefined') return null // Because Mapbox refuses to work with server-side rendering
+
+    const { classNames } = this.props
+    const ReactMapboxGl = require('react-mapbox-gl').default
+    const { Marker, Cluster } = require('react-mapbox-gl')
+
+    const Map = ReactMapboxGl({
+      accessToken: 'pk.eyJ1IjoiZWRoLXByb2ZzZXJ2IiwiYSI6ImNpb2o4aGNiaDAxeGp0eGttYXR0ZzVoMW8ifQ.Swi4xqTcQluPO0U1rFcJAg',
```
Comment (aabrook) [inline source/components/ClusterMap/index.js]: Should access tokens be baked into the code and not `'../../config/campaigns'`?

File: source/components/Metrics/index.js [hunk 4]
```diff
@@ -1,34 +1,63 @@
...
+    } = {},
+    leaderboards: {
+      individuals = [],
+      clubs = []
+    } = {}
+  } = {}
+}) => {
+  const raised = formatCents(reduce(charities, (acc, charity) => (
+    acc + charity.total_amount_cents
+  ), 0))
```
Comment (aabrook) [inline source/components/Metrics/index.js]:
```javascript
formatCents(_.sumBy(charities, _.property('total_amount_cents')))
```

File: source/components/ClusterMap/index.js [hunk 5]
```diff
@@ -1,138 +1,86 @@
...
-    }
+  const clusterMarker = (
+    coordinates,
+    clusterSize,
+    getIndividuals
+  ) => {
+    const count = numbro(getIndividuals().length).format('0,0')
+    const raised = formatCents(getIndividuals().reduce((acc, page) => (
+      acc + page.props.amount
+    )))
```
Comment (aabrook) [inline source/components/ClusterMap/index.js]:
```
formatCents(_.sumBy(getIndividuals(), _.property('page.props.amount')))
```

File: source/store/individuals/index.js [hunk 6]
```diff
@@ -1,20 +1,93 @@
...
+        fetchIndividualsSet(i + 2) // We need to skip page 1, it's already fetched
+      ))).then((individualsSet) => (
+        individualsSet.reduce((acc, { pages: pageSet }) => ([
+          ...acc,
+          ...pageSet
+        ]), [])
+      )).then((remainingIndividuals) => ([
+        ...individuals,
+        ...remainingIndividuals
+      ]))
```
Comment (aabrook) [inline source/store/individuals/index.js]: individuals as the starter above would stop the extra then

File: source/components/Metrics/index.js (hunk 4 above)
Comment (lukebrooker) [reply source/components/Metrics/index.js]: Nice!

Comment (ellsclytn) [reply source/components/Metrics/index.js]: Don't need `_.property`, as it turns out. Simply supplying a string as the second parameter to `_.subBy` uses it automatically 😃

Comment (aabrook) [reply source/components/Metrics/index.js]: Oooh that's nice to know!

**Pull request merged**